### PR TITLE
Add GetEffectiveRuleDetailsResponse & EffectiveRuleDetailsDto

### LINKED ISSUE #5193: Create model for the rule descriptions
#5162 has a stub model instead of actual model implementation

### PR DESCRIPTION
**Pull request by georgii-borovinskikh-sonarsource**
Fixes #5193

### CODE REVIEW

File: src/SLCore.UnitTests/SLCore.UnitTests.csproj [hunk 1]
```diff
@@ -13,4 +13,8 @@
     <ProjectReference Include="..\TestInfrastructure\TestInfrastructure.csproj" />
   </ItemGroup>
 
+  <ItemGroup>
+    <PackageReference Update="FluentAssertions" Version="6.12.0" />
```
Comment (georgii-borovinskikh-sonarsource) [inline src/SLCore.UnitTests/SLCore.UnitTests.csproj:17]: I updated it for only one project, but maybe I should have done it for https://github.com/SonarSource/sonarlint-visualstudio/blob/86d1c450a50aa31bb3390f9a124390f01fa73337/src/SonarLint.Test.props

Comment (ugras-ergun-sonarsource) [reply src/SLCore.UnitTests/SLCore.UnitTests.csproj:17]: Yes, you should do it there instead. We don't want to maintain in many places. Also why did you need to change version?

Comment (georgii-borovinskikh-sonarsource) [reply src/SLCore.UnitTests/SLCore.UnitTests.csproj:17]: I needed it for a feature that was not available in the old one

**Review by ugras-ergun-sonarsource — APPROVED** [review]
Please move the new version of Fluent Assertions to SonarLint.Test.props

**Pull request merged**